File: leetcode/cpp/1442_Count_Triplets_That_Can_Form_Two_Arrays_of_Equal_XOR.py
class Solution:
    def countTriplets(self, arr: list[int]) -> int:
        n = len(arr)
        ret = 0

        prefix = [0] * (n + 1)

        for i in range(n):
            prefix[i + 1] = prefix[i] ^ arr[i]

        for i in range(n):
            for k in range(i + 1, n):
                if prefix[i] == prefix[k + 1]:
                    ret += k - i

        return ret

File: leetcode/cpp/1442_Count_Triplets_That_Can_Form_Two_Arrays_of_Equal_XOR.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int countTriplets(vector<int>& arr) {
        int len = (int)arr.size();
        int ret = 0;

        vector<int> prefix(len + 1, 0);

        for (int i = 0; i < len; ++i)
            prefix[i + 1] = prefix[i] ^ arr[i];

        for (int i = 0; i < len; ++i) {
            for (int k = i + 1; k < len; ++k) {
                if (prefix[i] == prefix[k + 1])
                    ret += k - i;
            }
        }

        return ret;
    }
};
